AdNetworkSupport/MoPubMediationLogger.m

```diff
@@ -0,0 +1,124 @@
+{
+
+    self = [super init];
+    if(self)
+    {
+        self.NETWORK_TYPE=networkDictionary[@(networkType)];
+        self.ADFORMAT_TYPE=adFormatDictionary[@(adFormat)];
+        
+    }
+    return self;
+}
+
+//Type B implementation: Fetch network name and ad format from parsing the class names from the adapter
+// This process of fetching data is cumbersome and requires to build parsing logic
+-(instancetype)initWithClassName:(NSString *)className
+{
+    
+    self = [super init];
+
+    if(self)
+    {
+        [MoPubMediationLogger load];
+        [self parseSdkAndAdFormat:className];
+    }
+    return self;
+}
+
+-(void)parseSdkAndAdFormat:(NSString *)className
+{
+    for(NSString * item in adFormatArray){
+        NSArray *substrings = [className componentsSeparatedByString:item];
+        if(substrings.count==2){
+            self.NETWORK_TYPE= [substrings objectAtIndex:0];
+            self.ADFORMAT_TYPE=item;
+            break;
+        }
+    }
+    if(!self.NETWORK_TYPE){
+          self.NETWORK_TYPE = @"UNKOWN NETWORK";
+    }
+    if(!self.ADFORMAT_TYPE){
+        self.ADFORMAT_TYPE = @"UNKOWN AD Format";
+    }
+}
+
+//Log method to log network name, adformat and details of the log
+- (void)log :(Event) eventKey
+{
+    MPLogInfo(@"%@-%@-%@", self.NETWORK_TYPE,self.ADFORMAT_TYPE,eventDictionary[@(eventKey)]);
+    
+}
+
++ (void) load
+{
+    eventDictionary = @{
+                        @(AD_REQUESTED) : @"Requested.",
+                        @(AD_ERROR) : @"Error. Check logs for the error's details.",
+                        @(AD_UNAVAILABLE) : @"Not loaded/cached. Try again.",
+                        @(AD_WILLSHOW) : @" Will show the ad.",
+                        @(AD_SHOWN) : @"Now playing/showing.",
+                        @(AD_LOADED) : @"Cached/Loaded.",
+                        @(AD_IMPRESSED) : @"Logged an impression.",
+                        @(AD_CLICKED) : @"Clicked.",
+                        @(AD_DISMISSED) : @"Dismissed.",
+                        @(AD_WILLDISMISS) : @"Will dissis the ad.",
+                        @(AD_EXPIRED) : @"Expired",
+                        @(AD_COMPLETED) : @"Completed playing."
+                        };
+    
+    networkDictionary = @{
+                          @(AdColony) : @"AdColony",
+                          @(AppLovin): @"AppLovin",
+                          @(AdMob) : @"AdMob",
+                          @(Chartboost) : @"Chartboost",
+                          @(Facebook) : @"FacebookAudienceNetwork",
+                          @(Flurry) : @"Flurry",
+                          @(IronSource) : @"IronSource",
+                          @(OneByAOL) : @"OneByAOL",
+                          @(Tapjoy) : @"Tapjoy",
+                          @(UnityAds) : @"UnityAds",
+                          @(Vungle) : @"Vungle"
+                          };
+    
+    adFormatDictionary =@{
```

Comment:
Also if you let adapters pass in ad formats, you won't need this enum.

Reply:
It depends on what approach we are taking, type A or type B. Depending on the approach, we can delete the other form of implementation.